distribution/push_v2.go

```diff
@@ -397,8 +416,9 @@ func (pd *v2PushDescriptor) Upload(ctx context.Context, progressOutput progress.
 			return distribution.Descriptor{}, retryOnError(err)
 		}
 	}
-	defer layerUpload.Close()
-
+	if layerUpload != nil {
```

Comment:
It doesn't seem right to allow `layerUpload` to be nil and then immediately afterwards call `pd.uploadUsingSession` with `layerUpload`. Why would `layerUpload` be nil when `bs.Create` returned no error a few lines above?

Reply:
yes, this is paranoid, when I test the code with mock method I thought that may be the case when error is empty while layerUpload is nil too, and that should not happened